### DISCUSSION TITLE
How to vertically and horizontally align all images in montage operation?

### DISCUSSION THREAD

**Discussion by pvonmoradi** [ImageMagick/ImageMagick#4890] [Help] [answered]
Doesn't `-gravity center` make the images centrally aligned?
```
convert -size 256x256 xc:blue a.png
convert -size 512x256 xc:red b.png
convert -size 512x512 xc:green c.png
convert -size 112x700 xc:cyan d.png
montage a.png b.png c.png d.png -mode concatenate \
      -background none -tile 2x2 -geometry +5+5 -gravity center \
      out.png
```
Result:  
![image](https://user-images.githubusercontent.com/1058151/155416841-4035718f-2bac-499b-aa6c-d9737bb46338.png)

Expected:  
![image](https://user-images.githubusercontent.com/1058151/155417231-8cebff79-195f-493a-a518-24f7baa73c64.png)

(Ignore the white background. It is transparent in the command above)

```
Linux
Version: ImageMagick 6.9.7-4 Q16 x86_64 20170114 http://www.imagemagick.org
```

**Comment by fmw42** [comment, marked as answer]
try removing -mode concatenate. if that fails post your individual input images.

**Reply by pvonmoradi** [reply]
@fmw42 The images are generated from the convert commands above. Here they are: https://transfer.sh/pVwKtx/archive.zip
Removing `-mode concatenate` did not help

**Reply by fmw42** [reply]
I have never seen montage do what you are getting now. Perhaps something has changes.  But you can get around it by

```
convert -size 256x256 xc:blue a.png
convert -size 512x256 xc:red b.png
convert -size 512x512 xc:green c.png
convert -size 112x700 xc:cyan d.png
convert a.png b.png c.png d.png -gravity center -background none -extent 512x512 miff:- |\
montage - +repage \
-background none -tile 2x2 -geometry +5+5 \
out.png

```

**Reply by pvonmoradi** [reply]
@fmw42 Thanks, it worked
What does miff and extent do ? Also, did you mean my original montage command supposed to work as I expect? (like in v7)

**Reply by fmw42** [reply]
I got the same result as you in IM 6 and IM 7.  In the past it used to center as you thought it would. But I am not able to make it do that. I suspect a bug. 

-extent extends the image canvas to the size specified.  Miff is an internal Imagemagick format. I use convert to extend all the images to the same 512x512 size and then pipe (pass) the results to montage so that you do not need and extra command and a temporary image.
